main.rb

```diff
@@ -0,0 +1,120 @@
+module Enumerable
+  # each
+  def my_each
+    each do |i|
+      yield i
+    end
+  end
+
+  # each_with_index
+  def my_each_with_index
+    result = map do |element, index|
+      yield "#{index}: #{element} "
+    end
+    result
+  end
+
+  # select
+  def my_select
+    result = []
+    each do |i|
+      result << i if yield(i)
+    end
+    result
+  end
+
+  # all
+  def my_all?
+    my_each do |_elememt|
+      return false unless yield(element)
+    end
+    true
+  end
+
+  # any
+  def my_any?
+    my_each do |_elememt|
+      return true if yield(element)
+    end
+    false
+  end
+
+  # negated
+  def my_none?
+    my_each do |element|
+      yield(element)
+    end
+  end
+
+  # count
+  def my_count(element = false)
+    return length if elememt == false && !block_given?
+
+    arr = []
+    if element
+      to_a.my_each do |e|
+        arr << e if e == element
+      end
+    else
+      to_a.my_each do |e|
+        arr << e if yield(e)
+      end
+    end
+    arr.length
+  end
```

Comment:
- [ ] `my_count` returns the number of items in enum through enumeration
`[1, 2, 4, 7, 3, 8, 10, 13, 12, 99].my_count` returns 10

- [ ] `my_count` counts the number of items in enum that are equal to item if an argument is given
`[1, 2, 4, 1, 3, 8, 1, 13, 12, 1].my_count(1)` returns 4

- [ ] `my_count` counts the number of elements yielding a true value if a block is given
`[4, 5, 8, 4, 6, 9, 10].count { |num| num > 6 }` returns 3